### ISSUE TITLE
WiFi Temperature & Humidity Sensor reporting "Unknown" Humidity when Xtended Tuya is enabled 

### ISSUE THREAD

**Issue by pansament** [azerty9971/xtend_tuya#150]
Long story short when I restart Home Assistant instance for 1 minute Humidity sensor reports ok but after that minute is reporting "Unknown".

I m running HAOS with Core version 2024.10.4 with Xtend Tuya version v3.0.5-beta.1.
Problem apeared on latest non beta version so I tried beta version to see if the problem is fixed and no success.
The sensor is tuya temperature and humidity with lcd powered by 3 AAA bateries .

**Comment by pansament** [comment]
> Hello, Can you try with the latest BETA version please?
> 
> It should hopefully fix this issue :)

Updated from v3.0.5-beta.1 to v3.0.5-beta.2 and 10 minutes after Home Assistant restart, humidity sensor reports ok. 
Thank you for resolving this isue.

**Comment by Ramon1347** [comment]
> Hello, Can you try with the latest BETA version please?
> 
> It should hopefully fix this issue :)

Update: it shows the value of the humidity, but it doesn't update automatically like the temperature. I had to load the xtended tuya integration new to get the new values for the humidity

**Comment by pansament** [comment]
> > Hello, Can you try with the latest BETA version please?
> > It should hopefully fix this issue :)
> 
> Update: it shows the value of the humidity, but it doesn't update automatically like the temperature. I had to load the xtended tuya integration new to get the new values for the humidity

Same here. :(

**Comment by azerty9971** [comment]
Hello,
Can you try with the latest stable release please?

Thanks in advance,
Azerty

**Comment by pansament** [comment]
> Hello, Can you try with the latest stable release please?
> 
> Thanks in advance, Azerty

Updated but no refresh. Restarted home assistant several times but it is not refreshing automaticly.

**Comment by azerty9971** [comment]
Is this still happening in the latest stable version?
By any chance, do you have warnings in your system logs regarding this device? (settings -> system -> logs)

Have a nice day,
Azerty

**Comment by adrian-bulicanu** [comment]
Hi Azerty,

First of all, I want to thank you for your effort. It’s definitely a must-have extension for all Tuya users.

I'm experiencing the same issue, even on the latest versions: HA 2024.12.3 and Xtend Tuya 3.0.7. Here are the relevant logs, with the error at the end:

`2024-12-15 18:32:25.759 DEBUG (Thread-4 (_thread_main)) [tuya_sharing] _on_log: Received PUBLISH (d0, q0, r0, m0), 'cloud/device/bfe2a0bdc084a51064sa9o/in/24d2afb98121974b6505c40a3a980362/pen', ...  (285 bytes)
2024-12-15 18:32:25.759 DEBUG (Thread-4 (_thread_main)) [tuya_sharing] payload-> b'{"protocol":4,"data":{"devId":"bfe2a0bdc084a51064sa9o","dataId":"26d3af7e-c415-4b83-a859-77afd08e26e0","productKey":"vadjntdjedbnne0h","status":[{"dpId":1,"t":1734280345430,"value":228},{"dpId":2,"t":1734280345430,"value":51},{"dpId":4,"t":1734280345430,"value":0}]},"t":1734280345729}'
2024-12-15 18:32:25.759 DEBUG (Thread-4 (_thread_main)) [tuya_sharing] on_message: {'protocol': 4, 'data': {'devId': 'bfe2a0bdc084a51064sa9o', 'dataId': '26d3af7e-c415-4b83-a859-77afd08e26e0', 'productKey': 'vadjntdjedbnne0h', 'status': [{'dpId': 1, 't': 1734280345430, 'value': 228}, {'dpId': 2, 't': 1734280345430, 'value': 51}, {'dpId': 4, 't': 1734280345430, 'value': 0}]}, 't': 1734280345729}
2024-12-15 18:32:25.759 DEBUG (Thread-4 (_thread_main)) [tuya_sharing] mq receive-> {'protocol': 4, 'data': {'devId': 'bfe2a0bdc084a51064sa9o', 'dataId': '26d3af7e-c415-4b83-a859-77afd08e26e0', 'productKey': 'vadjntdjedbnne0h', 'status': [{'dpId': 1, 't': 1734280345430, 'value': 228}, {'dpId': 2, 't': 1734280345430, 'value': 51}, {'dpId': 4, 't': 1734280345430, 'value': 0}]}, 't': 1734280345729}
2024-12-15 18:32:25.759 DEBUG (Thread-4 (_thread_main)) [tuya_sharing] mq _on_device_report-> [{'dpId': 1, 't': 1734280345430, 'value': 228, 'code': 'va_temperature'}, {'dpId': 2, 't': 1734280345430, 'value': 51, 'code': 'humidity_value'}, {'dpId': 4, 't': 1734280345430, 'value': 0, 'code': 'battery_percentage'}]
2024-12-15 18:32:25.759 DEBUG (Thread-4 (_thread_main)) [tuya_sharing] mq _on_device_report before strategy convert strategy_name=default,dp_item=('va_temperature', 228),config_item={'statusFormat': '{"va_temperature": "$"}', 'valueDesc': '{"unit": "\\u2103", "min": -200, "max": 600, "scale": 1, "step": 1}', 'valueType': <DPType.INTEGER: 'Integer'>, 'enumMappingMap': {}, 'pid': 'vadjntdjedbnne0h'}
2024-12-15 18:32:25.759 DEBUG (Thread-4 (_thread_main)) [tuya_sharing] mq _on_device_report after strategy convert code=va_temperature,value=228
2024-12-15 18:32:25.760 DEBUG (Thread-4 (_thread_main)) [tuya_sharing] mq _on_device_report before strategy convert strategy_name=default,dp_item=('humidity_value', 51),config_item={'statusFormat': '{}', 'valueDesc': '{"unit": "%", "min": 0, "max": 100, "scale": 0, "step": 1}', 'valueType': <DPType.INTEGER: 'Integer'>, 'enumMappingMap': {}, 'pid': 'vadjntdjedbnne0h'}
2024-12-15 18:32:25.760 ERROR (Thread-4 (_thread_main)) [tuya_sharing] on message error = 'popitem(): dictionary is empty'`

**Comment by Ramon1347** [comment]
Hey, i had the latest beta and it worked well with the automatically update of the humidity. Until the 16.12.2024. I don't know what change there, i think i updated to the latest version (3.0.7) and since that day the humidity does not update automatically anymore.

Best regards

**Comment by dahadd** [comment]
3.0.7. humidity is not working

**Comment by pansament** [comment]
Hello again. How can I help you to diagnose this isue? Half of the isue is solved... Not reporting "Unknown" any more but is frustrating that is not auto refreshing the value. The value is read if I restart Home Assistant. But no auto import value from tuya after first read. 
Or how can I restrict Xtended Tuya to intercept that entities and for those get value from original tuya integration?

**Comment by pansament** [comment]
> Hello, Can you retry with v3.0.8-beta.2?
> 
> Also, can you export the device diagnostic so that I can have a look at it (using the latest version)
> 
> Thanks in advance, Azerty

Updated to v3.0.8-beta.2

Atachet 2 files of diagnostics. One from tuya one from xtend tuya.

I hope it helps... 

Here if you need more



[xtend_tuya-01JGE5DC9R36KQDQEAC6V9AGAW-Out-5de145b8c08be6d6e3c15dbeb81be1bf.json](https://github.com/user-attachments/files/18551828/xtend_tuya-01JGE5DC9R36KQDQEAC6V9AGAW-Out-5de145b8c08be6d6e3c15dbeb81be1bf.json)
[tuya-01JCAP7DAPBFCSMWRQ3EAYEGYE-Out-5de145b8c08be6d6e3c15dbeb81be1bf.json](https://github.com/user-attachments/files/18551827/tuya-01JCAP7DAPBFCSMWRQ3EAYEGYE-Out-5de145b8c08be6d6e3c15dbeb81be1bf.json)

**Comment by azerty9971** [comment]
Hello,
Can you share this device with me please?
I'd like to see what the API return...

From Tuya/Smartlife, you can share it with:
EU:
devun999999@gmail.com
US:
devun999999.us@gmail.com

(and let me know if you use Tuya and if you are in EU or US please :))

Thanks in advance,
Azerty

**Comment by azerty9971** [comment]
sure, https://meet.google.com/vkm-hpip-hkn

be right back in 2 minutes ;)

**Comment by pansament** [comment]
Hello...

Updated to v3.0.8-beta.5 and after HA restart Humidity apears for one minute then is reporting Unknown again.

**Comment by pansament** [comment]
> Here we are again, Humidity missing :D i try the latest beta version now and report back.
> 
> We had this some months ago, remember?
> 
> Edit: received new values after update, lets see for how long.

I m using v3.0.8 and i dont have this issue. 
I did not had time to test newer versions and colaborate with azerty9971 to eliminate this issue... 
So for now you can download v3.0.8

**Comment by Maxxxel** [comment]
Aight, downgrading now.

**Comment by Maxxxel** [comment]
Update: as soon as values want to update on change, it turns into unknown state.

We had this exact error months ago.

**Comment by pansament** [comment]
> Update v.3.0.8: as soon as values want to update on change, they turn into unknown state.
> 
> We had this exact error months ago.

Did you restart HomeAssistant after redownload v3.0.8?

**Comment by Maxxxel** [comment]
sure

**Comment by pansament** [comment]
This is wierd....
On my setup is working great on that version. 

![Image](https://github.com/user-attachments/assets/24ff3ff6-a07b-4ad7-bca4-dbc4b8cbfb20)
![Image](https://github.com/user-attachments/assets/db837fd9-a0be-41cc-a385-1be567a8e6d9)
![Image](https://github.com/user-attachments/assets/857e916d-8d11-4d1d-b4ef-94a812a94cf6)

**Comment by pansament** [comment]
v3.0.11-beta.5 - Reports "Unknown" after 1 minute after restart
v3.0.11-beta.4 - Reports "Unknown" after 1 minute after restart
v3.0.11-beta.3 - Reports "Unknown" after 1 minute after restart
v3.0.11-beta.1 - Reports "Unknown" after 1 minute after restart
v3.0.10            - Reports "Unknown" after 1 minute after restart
v3.0.9              - Working and update value when Humidity changes on thermomether. Not reporting "Unknown" after 1 minute restart
v3.0.0              - Working and update value when Humidity changes on thermomether. Not reporting "Unknown" after 1 minute restart
v3.0.8-beta.5 - Reports "Unknown" after 1 minute after restart


So I will use v3.0.9 because is latest version that reports humidity and not unknown value.

**Comment by azerty9971** [comment]
Hello,
The latest stable version should address this issue :)

Can you try it and let me know? (3.0.12)

Thanks in advance,
Azerty

**Comment by pansament** [comment]
> Hello, The latest stable version should address this issue :)
> 
> Can you try it and let me know? (3.0.12)
> 
> Thanks in advance, Azerty

Hello,
After update from v3.0.9 I got "Unknown" status of humidity.
Renoved integration and uninstall integration.
Restart HA and reinstall integration v3.0.12 and fot 10 minutes looks ok.
I'll report tomorow.

**Comment by pansament** [comment]
Another day another post. I hope it's not considered spamming.

After last night's update the humidity sensor worked for about an hour.

Now that I saw that it reports an "Unknown" value I tried to go back to version 3.0.9 which I had until last night and it worked.

I deleted the integration from both integrations and HACS restarted Home Assistant and downloaded version 3.0.9 and after restart added the integration to Home assistant. The result was that Xtend Tuya no longer has devices or entities.

I restored a backup of Home Assistant made before the update to 3.0.12 I reauthenticated the tuya integration but still no devices or entities appeared in the Xtend Tuya integration.

Deleted and reinstalled version 3.0.9 but without favorable results.
Deleted and installed version 3.0.12 but the humidity sensor reports the value "Unknown" even after using OpenAPI credentials.

So long story short.... I have now lost the functionality of the humidity sensors in the Tuya integration.

**Comment by azerty9971** [comment]
Hello,
Don't worry, it's not considered spam ;)

I'm sorry to see that it didn't work as expected

Can you test with the latest beta? (3.1.0-beta.2)

Fingers crossed but it should in theory fix this issue...

And even if it fixes it, can you post the device diagnostic here so that I can see the final result?

Thanks!
Azerty

**Comment by Maxxxel** [comment]
I removed integrated Tuya and restarted everything 

Works as it should 😊

**Comment by azerty9971** [comment]
Actually on the latest version you don't have to remove the integrated Tuya (you can if you want to, but it's not mandatory anymore)

**Comment by pansament** [comment]
> Hello, Don't worry, it's not considered spam ;)
> 
> I'm sorry to see that it didn't work as expected
> 
> Can you test with the latest beta? (3.1.0-beta.2)
> 
> Fingers crossed but it should in theory fix this issue...
> 
> And even if it fixes it, can you post the device diagnostic here so that I can see the final result?
> 
> Thanks! Azerty

Hello.

Sorry for the delay in responding.

I haven't had time to test it yet.

Now I've deleted the integration from both HomeAssistant and HACS and restarted Home Assistant. I had version 3.0.12.
After the restart I installed version 3.1.0-beta.2 and restarted Home assistant and added the integration to Home Assistant and another restart to Home Assistant.
For about a minute I also had values ​​from the humidity sensors but then "Unknown" appeared again for both temperature and humidity sensors from Tuya.
In the Home Assistant logs nothing related to Tuya or Xtend Tuya appears.

I'm attaching the diagnostic file.

[config_entry-xtend_tuya-01JSY0FWTGYRSRMCDE761624YG.json](https://github.com/user-attachments/files/19938781/config_entry-xtend_tuya-01JSY0FWTGYRSRMCDE761624YG.json)

**Comment by azerty9971** [comment]
Hello,
Can you try with the latest beta? (3.1.0-beta.3)

It should fix this :)

Thanks in advance,
Azerty

**Comment by pansament** [comment]
> Hello, Can you try with the latest beta? (3.1.0-beta.3)
> 
> It should fix this :)
> 
> Thanks in advance, Azerty

Here I would like to share that after 12 hours of updating to version 3.1.0-beta.3, the humidity sensors and the rest of the Xtend Tuya integration are working perfectly! 

Thanks for the support!

**Comment by azerty9971** [comment]
Hello,
This issue should be resolved in v4.0.0.

If it's not the case please reopen this issue.

Have a nice day,
Azerty
